src/tools/illink/src/linker/Linker/TypeMapInfo.cs

```diff
@@ -293,6 +294,18 @@ void FindAndAddDefaultInterfaceImplementations (TypeDefinition type, MethodDefin
 				if (potentialImplInterface == null)
 					continue;
 
+				// If the interface doesn't implement the interface with the method we're looking for, exit early
+				bool potentialImplInterfaceImplementsInterface = false;
+				foreach (var iface in potentialImplInterface.Interfaces) {
+					if (context.TryResolve (iface.InterfaceType) == interfaceMethod.DeclaringType)
+					{
+						potentialImplInterfaceImplementsInterface = true;
+						break;
```

Comment:
You could refactor this into a local function to avoid doing the `break`-`continue` dance.